### DISCUSSION TITLE
[1일차-예진] 클로저

### DISCUSSION THREAD

**Discussion by ez1n** [zzangCS/zzangInterview#5] [프론트 면접 질문]
클로저의 정의와 사용하는 이유에 대해 설명해보세요.

**Comment by ez1n** [comment]
정의 : 어떤 함수가 자신의 내부가 아닌 외부에서 선언된 변수에 접근하는 것
사용 이유 : 현재 상태를 기억하고 변경된 최신상태를 유지할 수 있으며 전역변수의 사용을 억제하고 정보를 은닉하기 위해 사용됩니다.

**Comment by smarfy99** [comment]
클로저는 함수와 그 함수가 선언된 환경과의 관계의 개념이다.
클로저는 스코프로 변수의 접근 범위를 한정하는 것이 핵심이다.

**Reply by smarfy99** [reply]
https://hanamon.kr/javascript-%ED%81%B4%EB%A1%9C%EC%A0%80/

**Comment by skyhanull** [comment]
즉, 클로저는 반환된 내부함수가 자신이 선언됐을 때의 환경(Lexical environment)인 스코프를 기억하여 자신이 선언됐을 때의 환경(스코프) 밖에서 호출되어도 그 환경(스코프)에 접근할 수 있는 함수
를 말한다. 이를 조금 더 간단히 말하면 클로저는 자신이 생성될 때의 환경(Lexical environment)을 기억하는 함수다
